package/base-files/files/etc/rc.common

```diff
@@ -55,7 +55,12 @@ enable() {
 
 enabled() {
 	name="$(basename "${initscript}")"
-	[ -x "$IPKG_INSTROOT/etc/rc.d/S${START}${name##S[0-9][0-9]}" ]
+	name="${name##[SK][0-9][0-9]}"
+	{
+		[ -z "${START:-}" ] || [ -x "$IPKG_INSTROOT/etc/rc.d/S${START}$name" ]
```

Comment:
Aren't the `[ -z "${START:-}" ]` and `[ -z "${STOP:-}" ]` conditions always true and due to that effectively redundant? What's the intention behind the check? Wouldn't a mere `[ -x "$IPKG_INSTROOT/etc/rc.d/S${START}$name" ] || [ -x "$IPKG_INSTROOT/etc/rc.d/K${STOP}$name" ]` suffice?

Btw., do you think it makes sense to check for `-L` rather than `-x` ? The enable/disable action manipulates symlinks after all so it might make sense to peg the enabled status to the expected path being a symlink.

Reply:
The first part of that condition is there to cover the case when `START` is not defined. In such case the `${START:-}` is expanded to an empty string and thus condition is successful. The result would be a failure without it as there is going to be clearly no link `/etc/rc.d/Sfoo` for service `foo` that does not define `START` variable.

Yeah, it can be check for links. It creates links and it pretty much highly depends on that those are links so we can check for symlinks.

Reply:
Ah right, I misread `${START:-}` as `${START:--}`. If there's no replacement string, is there any reason to not simply use `[ -z "$START" ]` ?

Reply:
Well right now it doesn't matter but it would be nice to once in the future use `set -u` in init scripts to cover common type errors. The `${START:-}` is expanded to an empty variable even if `START` is not defined while `$START` raises an error in the case of `set -u`.